### PR TITLE
Update the scientific Python stack

### PR DESCRIPTION
**Pull request by bgruening**
This will update our entire Python stack to the latest versions. Moreover, all packages are depending on an own Python version. This will make it possible to make the next version of this stack depend on Python-3.5/3.6.

We need this patch to make it work:
https://github.com/galaxyproject/galaxy/pull/698

And you need to set `enable_beta_tool_command_isolation = True`. I would like to get this into the TTS as soon as possible to test this extensively before the next Galaxy release.

### CODE REVIEW

**Comment by bgruening** [comment]
Ok, ready for review.
A complete new python-sci stack and a few random changes.

File: packages/package_python_2_7/tool_dependencies.xml [hunk 1]
```diff
@@ -45,17 +45,17 @@
...
           </repository>
         </action>
         <!-- by default python is not looking af LD_INCLUDE_PATH and Co. -->
         <action type="shell_command">sed -i.bak -e "s/if host_platform == 'atheos':/if True:/g" setup.py</action>
 
         <action type="autoconf"/>
-        <action type="download_file">https://bitbucket.org/pypa/setuptools/get/6.1.tar.bz2</action>
-        <action type="shell_command">tar -xjf 6.1.tar.bz2</action>
-        <action type="change_directory">pypa-setuptools-7ea80190d494</action>
+        <action type="download_file">https://bitbucket.org/pypa/setuptools/get/18.2.tar.bz2</action>
```
Comment (erasche) [inline packages/package_python_2_7/tool_dependencies.xml]: sha sum

File: packages/package_python_2_7_10/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            <package name="gdbm" version="1.11" />
+          </repository>
+          <repository name="package_ncurses_6_0" owner="iuc">
+            <package name="ncurses" version="6.0" />
+          </repository>
+        </action>
+        <!-- by default python is not looking af LD_INCLUDE_PATH and Co. -->
+        <action type="shell_command">sed -i.bak -e "s/if host_platform == 'atheos':/if True:/g" setup.py</action>
+        <action type="autoconf" />
+        <action type="download_file">https://bitbucket.org/pypa/setuptools/get/18.2.tar.bz2</action>
```
Comment (erasche) [inline packages/package_python_2_7_10/tool_dependencies.xml]: sha

**Comment by erasche** [comment]
Okay, this all looks OK to me. We'll go back and update to sha256sums later, since that's not so important.

**Pull request merged**